### PR TITLE
Update to React18 and Fluent9 and other fixes

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
**Change Description**:

This does for the JavaScript React template the same things that [128](https://github.com/OfficeDev/Office-Addin-TaskPane-React/pull/128) does for the TypeScript React template (except for the things that are only relevant to TypeScript). So please see that PR's description. 

In addition, this PR does the following:

1. Fixes [104](https://github.com/OfficeDev/Office-Addin-TaskPane-React-JS/issues/104)
2. Corrects the file extension on the React component files from "js" to "jsx".
3. Adds "allowJs" to the tsconfig.json file to fix a problem with the unit tests.
4. Removes some duplicate entries in the tasks.json file.
5. Removes a duplicate section from the Readme file. 

1. **Do these changes impact any *npm scripts* commands (in package.json)?** (e.g., running 'npm run start')
    If Yes, briefly describe what is impacted.
    No

2. **Do these changes impact *VS Code debugging* options (launch.json)?**
    If Yes, briefly describe what is impacted.
    No

3. **Do these changes impact *template output*?** (e.g., add/remove file, update file location, update file contents)
    If Yes, briefly describe what is impacted.
    Yes. See **Change Description** above.

4. **Do these changes impact *documentation*?** (e.g., a tutorial on https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins)
    If Yes, briefly describe what is impacted.
    Yes. The quickstart/tutorial for a React-based add-in will have to be changed and possibly other files too. **THIS PR SHOULD NOT BE MERGED TO PRODUCTION UNTIL THE DOC REVISIONS ARE READY TO BE MERGED TOO. NEEDS TO HAPPEN AT THE SAME TIME.**

If you answered yes to any of these please do the following:
    > Include 'Rick-Kirkham' in the review
    > Make sure the README file is correct

**Validation/testing performed**:

All unit and end2end tests work locally.
All emitted projects tested and they all work.
`npm start` works on the pre-converted project.

### CODE REVIEW

**Review by millerds — COMMENTED** [review]
I tried to run the tests locally in the PR, but something went wrong with the first test and blocked all the rest.

File: README.md [hunk 1]
```diff
@@ -27,13 +27,6 @@ Get a free sandbox, tools, and other resources you need to build solutions for t
 - [Access to experts](https://developer.microsoft.com/microsoft-365/dev-program#Experts) Access community events to learn from Microsoft 365 experts.
 - [Personalized recommendations](https://developer.microsoft.com/microsoft-365/dev-program#Recommendations) Find developer resources quickly from your personalized dashboard.
 
-## Join the Microsoft 365 Developer Program
-Get a free sandbox, tools, and other resources you need to build solutions for the Microsoft 365 platform.
-- [Free developer sandbox](https://developer.microsoft.com/microsoft-365/dev-program#Subscription) Get a free, renewable 90-day Microsoft 365 E5 developer subscription.
-- [Sample data packs](https://developer.microsoft.com/microsoft-365/dev-program#Sample) Automatically configure your sandbox by installing user data and content to help you build your solutions.
-- [Access to experts](https://developer.microsoft.com/microsoft-365/dev-program#Experts) Access community events to learn from Microsoft 365 experts.
-- [Personalized recommendations](https://developer.microsoft.com/microsoft-365/dev-program#Recommendations) Find developer resources quickly from your personalized dashboard.
-
```
[the remark this answers is not in the document]
Comment (Rick-Kirkham) [reply README.md:36]: > Is there a reason we're removing this?

Yes. The exact same section appears twice in the Readme.

[the remark this answers is not in the document]
Comment (millerds) [reply README.md:36]: I see it now.

File: convertToSingleHost.js [hunk 2]
```diff
@@ -167,15 +123,15 @@ async function updateLaunchJsonFile() {
 }
 
 function getHostName(host) {
-  switch (host) {
+  switch(host) {
```
[the remark this answers is not in the document]
Comment (Rick-Kirkham) [reply convertToSingleHost.js]: The linter doesn't object, but I added the space anyway.

**Comment by Rick-Kirkham** [comment]
> I tried to run the tests locally in the PR, but something went wrong with the first test and blocked all the rest.

First unit test or first e2e test? What are the exact symptoms? Is there an error message or does it just hang? If the latter, at what point?

**Comment by millerds** [comment]
> > I tried to run the tests locally in the PR, but something went wrong with the first test and blocked all the rest.
> 
> First unit test or first e2e test? What are the exact symptoms? Is there an error message or does it just hang? If the latter, at what point?

My machine must have been in a bad state . . . tests worked after a reboot.

**Review by millerds — APPROVED** [review]

**Review by millerds — APPROVED** [review]

**Pull request merged**